cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```
@@ -74,12 +130,66 @@ public void schedule(Id.Program program, SchedulableProgramType programType, Sch
+   *                     or -1 if it never happened yet
+   * @param basePollTs timestamp, in milliseconds, which matches the time at which {@code basePollSize} was computed.
+   *                   -1 indicates to start counting from the current timestamp
+   * @param persist {@code true} if this schedule should be persisted in the persistent store containing the
+   *                stream size schedules, {@code false} otherwise
+   */
+  private void schedule(Id.Program program, SchedulableProgramType programType, StreamSizeSchedule streamSizeSchedule,
+                        boolean active, long basePollSize, long basePollTs, boolean persist) throws SchedulerException {
+    // Create a new StreamSubscriber, if one doesn't exist for the stream passed in the schedule
+    Id.Stream streamId = Id.Stream.from(program.getNamespaceId(), streamSizeSchedule.getStreamName());
+    StreamSubscriber streamSubscriber = streamSubscribers.get(streamId);
+    if (streamSubscriber == null) {
+      streamSubscriber = new StreamSubscriber(streamId);
+      StreamSubscriber previous = streamSubscribers.putIfAbsent(streamId, streamSubscriber);
+      if (previous == null) {
+        try {
+          streamSubscriber.start();
+        } catch (NotificationFeedException e) {
+          streamSubscribers.remove(streamId);
+          LOG.error("Notification feed error for streamSizeSchedule {}", streamSizeSchedule);
+          throw new SchedulerException(e);
+        } catch (NotificationFeedNotFoundException e) {
+          streamSubscribers.remove(streamId);
+          LOG.error("Notification feed does not exist for streamSizeSchedule {}", streamSizeSchedule);
+          throw new SchedulerException(e);
+        }
+      } else {
+        streamSubscriber = previous;
+      }
+    }
+
+    // Add the scheduleTask to the StreamSubscriber
+    if (streamSubscriber.createScheduleTask(program, programType, streamSizeSchedule,
```

not sure when this would return false. One case I know is if the schedule already exists. Then you should throw an exception, not silently ignore the attempted overwrite.